types/src/v0/impls/block/full_payload/payload.rs

```diff
@@ -64,7 +64,7 @@ impl Payload {
         self.read_ns_payload(&ns_payload_range)
     }
 
-    pub(crate) fn byte_len(&self) -> PayloadByteLen {
+    pub fn byte_len(&self) -> PayloadByteLen {
```

Comment:
I wonder why we need to make this `pub`.

@mrain are we using the new API correctly? I had a brief look at `parse_ns_table` and it seems that due to the types used and the return value it may be rather easy to use it wrongly in the code. We have an `NsTable` type so I'm wondering why we now also use `Vec<Range<usize>>` to represent an NsTable.